bpf/tests/pktcheck.h

```
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#pragma once
+
+#include <bpf/compiler.h>
+#include <bpf/builtins.h>
+#include <bpf/helpers.h>
+
+#include <lib/endian.h>
+
+#include <linux/byteorder.h>
+#include <linux/ip.h>
+#include <linux/in.h>
+
+static __always_inline
+int pktcheck__validate_ipv4(struct iphdr *ip4, __u8 nexthdr, __be32 saddr, __be32 daddr)
+{
+	if (ip4->protocol != nexthdr)
+		return -1;
+	if (ip4->saddr != saddr)
+		return -1;
+	if (ip4->daddr != daddr)
+		return -1;
+
+	/* Skip csum validation if options are present */
```

Do we have tests with IP options? If yes, I'd rather support additional packet lengths, than have the checksum value untested in those cases. If not, I'd rather return a failure if the IP header length is unexpected, so that we don't miss the checksum validation when we start using/testing options.

---

There's some tests with IP options, I didn't touch them (so they just use the hard-coded csum check).

We can easily explore adding support for IP options later on, but I wouldn't want to block on that. So I'll go with your suggestion of returning an error for now :+1:.

---

>I'll go with your suggestion of returning an error for now

ACK.